editors/neovim/Portfile

```diff
@@ -45,8 +45,12 @@ depends_lib             port:gettext \
 
 cmake.build_type        Release
 
-configure.args-append   -DUSE_BUNDLED=OFF \
-                        -DLUA_PRG=${prefix}/bin/luajit
+configure.args-append   -DLUA_PRG=${prefix}/bin/luajit
+
+pre-configure {
+    system -W ${worksrcpath} "cmake -S cmake.deps -B .deps -G \"Unix Makefiles\" -D CMAKE_BUILD_TYPE=Release -DUSE_BUNDLED=OFF -DUSE_BUNDLED_TS_PARSERS=ON"
+    system -W ${worksrcpath} "cmake --build .deps"
```

Comment:
This doesn't make use of all of the MacPorts settings that we expect ports to make use of. For example, it doesn't build for the user's desired build_arch or universal_archs or with the MacPorts default optimization flags. All of those details are handled in the cmake portgroups, but those are designed for ports that do only that and can't be easily mixed with ports that also need to do other build tasks.

Reply:
The desired number of parallel build jobs is another MacPorts setting that the above does not make use of.

I see that this port already uses the cmake portgroup. cmake has extensive capabilities for building subprojects as a part of a larger build. Why hasn't neovim made use of those facilities so that everything gets built using just the original top-level cmake invocation that the port already uses?

If the tree-sitter parsers will forever remain a separate build, can they be built in a separate portfile that the neovim port depends on? That separate portfile could then simply use the cmake portgroup as intended to get all expected MacPorts behaviors for free.

Reply:
Exactly my thoughts. I wasn't able to find any reasoning behind this kind of design yet, so I opened an issue there. Let's see what they think: https://github.com/neovim/neovim/issues/29042
Separate port could work, but it seems unnecessarily complicated to have an additional package that only a single other package ever uses and depends on. If this will be the way to go, I also wonder if this could be done as a subport.

Reply:
Quick summary of the discussion with upstream: An attempt to move all deps into the toplevel cmake was made in https://github.com/neovim/neovim/pull/22054, but dismissed because of technical reasons.
For the parsers specifically, there is a slight chance it might happen in the future though.
But for the current release, we have to work around it of course. I see the following possibilities:
* Use an approach similar to what is already implemented in the PR, using a pre-configure step and try to shove all the cmake args from the cmake port group in there manually as good as possible
* Split neovim in two packages. Debian uses a similar approach (neovim + neovim-runtime): https://salsa.debian.org/vim-team/neovim/-/tree/debian/experimental/debian Though they have a lot of custom logic there as well and splitting packages like this is also more common in debian.
* If possible without too much effort, try to patch neovim's toplevel cmake to also include the parsers.

It would be good to know which approach macports maintainers would prefer before I move forward. I am currently leaning a bit towards the third option (patching), assuming the patch can be made small and clear, since (I think) this may be easier to revert than a split package in case it becomes better integrated upstream in the future.

Reply:
I'd prefer the second option, but if you can make a clean and easy to adapt patch, option 3 is acceptable.

Reply:
@ryandesign @l2dy I pushed a possible solution with a patch, which applies to both the release and neovim-devel. Whether it is clean is subjective I would say. If you don't like it I will attempt to split split the package.